blob/blob.go

```diff
@@ -30,19 +30,23 @@ const MaxShareVersion = 127
 // basic stateless checks over it.
 func New(ns namespace.Namespace, blob []byte, shareVersion uint8) *Blob {
 	return &Blob{
-		NamespaceId:      ns.ID,
+		NamespaceId:      ns.ID(),
 		Data:             blob,
 		ShareVersion:     uint32(shareVersion),
-		NamespaceVersion: uint32(ns.Version),
+		NamespaceVersion: uint32(ns.Version()),
 	}
 }
 
 // Namespace returns the namespace of the blob
-func (b *Blob) Namespace() namespace.Namespace {
-	return namespace.Namespace{
-		Version: uint8(b.NamespaceVersion),
-		ID:      b.NamespaceId,
-	}
+func (b *Blob) Namespace() (namespace.Namespace, error) {
+	return namespace.NewFromBytes(b.RawNamespace())
+}
+
+// Namespace returns the namespace of the blob
+func (b *Blob) RawNamespace() []byte {
+	namespace := []byte{byte(b.NamespaceVersion)}
+	namespace = append(namespace, b.NamespaceId...)
+	return namespace
 }
 
 // Validate runs a stateless validity check on the form of the struct.
```

Comment:
Besides Validate, the node needs two more validate funcs. https://github.com/celestiaorg/celestia-node/blob/540192259c144ccbd24e45e34616a41389232a51/share/namespace.go#L114-L129.

We need this so we can restrict to only blobs(excluding reserved) and to only real data(excluding parity)

Reply:
Let me extract this out as a separate issue. I imagine there's some overlap with celestia-app and the validation there